db/document.go

```diff
@@ -76,12 +76,64 @@ type syncData struct {
 	removedRevisionBodyKeys map[string]string // keys of non-winning revisions that have been removed (and so may require deletion), indexed by revID
 }
 
+func (doc *SyncData) HashRedact(salt string) SyncData {
+
+	syncData := SyncData{
+		CurrentRev:      doc.CurrentRev,
+		NewestRev:       doc.NewestRev,
+		Flags:           doc.Flags,
+		Sequence:        doc.Sequence,
+		UnusedSequences: doc.UnusedSequences,
+		RecentSequences: doc.RecentSequences,
+		History:         RevTree{},
+		Channels:        channels.ChannelMap{},
+		Access:          UserAccessMap{},
+		RoleAccess:      UserAccessMap{},
+		Expiry:          doc.Expiry,
+		Cas:             doc.Cas,
+		Crc32c:          doc.Crc32c,
+		TombstonedAt:    doc.TombstonedAt,
+		Attachments:     AttachmentsMeta{},
+	}
+
+	for k, v := range doc.Channels {
+		syncData.Channels[base.Sha1HashString(k, salt)] = v
+	}
+
+	for k, v := range doc.History {
+		revInfo := *v
+
+		if revInfo.Channels != nil {
+			revInfo.Channels = base.Set{}
+			for existingChanKey := range v.Channels {
+				revInfo.Channels.Add(base.Sha1HashString(existingChanKey, salt))
+			}
+		}
+
+		syncData.History.addRevision(k, revInfo)
+	}
+
+	for k, v := range doc.Access {
+		syncData.Access[base.Sha1HashString(k, salt)] = v
```

Comment:
I think 'v' is a channel name here, which would need to get hashed as well. (same in the RoleAccess case below)